persistence/persistence-mybatis-test/src/test/java/org/cbioportal/persistence/mybatis/SampleMyBatisRepositoryTest.java

```diff
@@ -0,0 +1,289 @@
+        Assert.assertEquals("TCGA-A1-A0SB", patient.getStableId());
+        Assert.assertEquals((Integer) 1, patient.getCancerStudyId());
+        Assert.assertEquals("study_tcga_pub", patient.getCancerStudyIdentifier());
+        CancerStudy cancerStudy = patient.getCancerStudy();
+        Assert.assertEquals((Integer) 1, cancerStudy.getCancerStudyId());
+        Assert.assertEquals("study_tcga_pub", cancerStudy.getCancerStudyIdentifier());
+        Assert.assertEquals("brca", cancerStudy.getTypeOfCancerId());
+        Assert.assertEquals("Breast Invasive Carcinoma (TCGA, Nature 2012)", cancerStudy.getName());
+        Assert.assertEquals("BRCA (TCGA)", cancerStudy.getShortName());
+        Assert.assertEquals("<a href=\\\"http://cancergenome.nih.gov/\\\">The Cancer Genome Atlas (TCGA)</a> Breast" +
+                " Invasive Carcinoma project. 825 cases.<br><i>Nature 2012.</i> <a href=\\\"http://tcga-data.nci." +
+                "nih.gov/tcga/\\\">Raw data via the TCGA Data Portal</a>.", cancerStudy.getDescription());
+        Assert.assertEquals(true, cancerStudy.getPublicStudy());
+        Assert.assertEquals("23000897", cancerStudy.getPmid());
+        Assert.assertEquals("TCGA, Nature 2012", cancerStudy.getCitation());
+        Assert.assertEquals("SU2C-PI3K;PUBLIC;GDAC", cancerStudy.getGroups());
+        Assert.assertEquals((Integer)0 , cancerStudy.getStatus());
+    }
+
+    @Test
+    public void getAllSamplesOfPatientInStudySummaryProjection1PageSize() throws Exception {
+
+        List<Sample> result = sampleMyBatisRepository.getAllSamplesOfPatientInStudy("study_tcga_pub", "TCGA-A1-A0SB",
+                "SUMMARY", 1, 0, null, null);
+
+        Assert.assertEquals(1, result.size());
+    }
+
+    @Test
+    public void getAllSamplesOfPatientInStudySummaryProjectionStableIdSort() throws Exception {
+
+        List<Sample> result = sampleMyBatisRepository.getAllSamplesOfPatientInStudy("study_tcga_pub", "TCGA-A1-A0SB",
+                "SUMMARY", null, null, "stableId", "ASC");
+
+        Assert.assertEquals(2, result.size());
+        Assert.assertEquals("TCGA-A1-A0SB-01", result.get(0).getStableId());
+        Assert.assertEquals("TCGA-A1-A0SB-02", result.get(1).getStableId());
+    }
+
+    @Test
+    public void getMetaSamplesOfPatientInStudy() throws Exception {
+
+        BaseMeta result = sampleMyBatisRepository.getMetaSamplesOfPatientInStudy("study_tcga_pub", "TCGA-A1-A0SB");
+
+        Assert.assertEquals((Integer) 2, result.getTotalCount());
+    }
+
+    @Test
+    public void fetchSamples() throws Exception {
+
+        List<String> studyIds = new ArrayList<>();
+        studyIds.add("study_tcga_pub");
+        studyIds.add("study_tcga_pub");
+        List<String> sampleIds = new ArrayList<>();
+        sampleIds.add("TCGA-A1-A0SB-01");
+        sampleIds.add("TCGA-A1-A0SE-01");
+
+        List<Sample> result = sampleMyBatisRepository.fetchSamples(studyIds, sampleIds, "SUMMARY");
```

Comment:
This is for the cases that user enters list of sample ids in a big text area in the user interface, just like when you click `User Defined Case-List` in query page.